islandora_plupload.module

```diff
@@ -55,6 +55,33 @@ function islandora_plupload_alter_managed_files(&$form) {
       $element =& $form[$element_key];
       islandora_plupload_alter_managed_files($element);
       if (isset($element['#type']) && $element['#type'] === 'managed_file' && !(isset($element['#islandora_plupload_do_not_alter']) && $element['#islandora_plupload_do_not_alter'])) {
+        // The Plupload and Islandora Plupload modules create a bug that
+        // prevents users from adding datastreams and replacing datastreams that
+        // allow any file type as a result of validating a files extension
+        // (ex. .jpg) against an empty string.
+        $current_path = current_path();
```

Comment:
Since that's just a wrapper around `$_GET['q']`, it's missing all that drupal-y hook/alter goodness and might break in some cases. Should use `menu_get_item()` instead, that would also allow you to match directly against `$menu_item['path'] == 'islandora/object/%/manage/datastreams/add'`